OpenEdXMobile/src/main/java/org/edx/mobile/view/BulkDownloadFragment.java

```diff
@@ -502,18 +539,33 @@ public void onRevisit() {
     }
 
     public void onEvent(BulkVideosDownloadCancelledEvent event) {
+        if (getView() == null) {
+            // If fragment view is not created yet, mark the event and fire it once the view is created.
+            bulkDownloadWasCancelled = true;
+            return;
+        }
         binding.swDownload.setChecked(false);
         switchState = SwitchState.DEFAULT;
         prefManager.setBulkDownloadSwitchState(switchState, videosStatus.courseComponentId);
         updateUI();
     }
 
     public void onEvent(BulkVideosDownloadStartedEvent event) {
+        if (getView() == null) {
+            // If fragment view is not created yet, mark the event and fire it once the view is created.
+            bulkDownloadWasStarted = true;
+            return;
+        }
         remainingVideos.clear();
         updateUI();
     }
 
-    private class CourseVideosStatus {
+    private boolean allVideosDownloading(@NonNull CourseVideosStatus videosStatus) {
```

Comment:
Because the inner class implementing Parcelable must be static.

Reply:
This function should've been refactored/fixed instead of moving it out of the class which doesn't make any sense.

Here's how it can still be in the `CourseVideosStatus` class and work as expected:

```java
boolean allVideosDownloading(SwitchState switchState) {
    return (total == downloaded + downloading) || (switchState != null && switchState == SwitchState.USER_TURNED_ON);
}
```